src/App.vue

```diff
@@ -107,33 +129,166 @@ const resetComparison = () => {
+  comparisonResult.value.removed_rows.forEach(collectHeaders);
+  comparisonResult.value.changed_rows.forEach(collectHeaders);
+
+  // Convert the Set of headers into an Array and sort them
+  const headers = Array.from(allHeaders).sort();
+
+  // Ensure the comparisonHeader (e.g., "ID") is the first column
+  const comparisonHeader = selectedHeader.value;
+  const headerIndex = headers.indexOf(comparisonHeader);
+  if (headerIndex !== -1) headers.splice(headerIndex, 1);
+  headers.unshift(comparisonHeader);
+
+  // Add Source and Message as the last columns
+  headers.push("Source", "Message");
+
+  // Prepare CSV content with dynamic headers
+  let csvContent = [headers.join(",")];
+
+  // Helper function to create a CSV row from a record
+  const createCsvRow = (id, record, source, message) => {
+    return headers.map((header) => {
+      let value = "";
+
+      if (header === comparisonHeader) {
+        value = id || "";
+      } else if (header === "Source") {
+        value = source || "";
+      } else if (header === "Message") {
+        value = message || "";
+      } else {
+        value = record[header] || "";
+      }
+
+      // Escape commas and quotes for CSV
+      if (value.includes(",") || value.includes('"')) {
+        value = `"${value.replace(/"/g, '""')}"`;
+      }
+
+      return value;
+    }).join(",");
+  };
+
+  // Iterate over added rows
+  for (const row of comparisonResult.value.added_rows) {
+    const csvRow = createCsvRow(
+      row.id,
+      row.record,
+      newFileName,
+      "Record only in new file"
+    );
+    csvContent.push(csvRow);
+  }
+
+  // Iterate over removed rows
+  for (const row of comparisonResult.value.removed_rows) {
+    const csvRow = createCsvRow(
+      row.id,
+      row.record,
+      oldFileName,
+      "Record only in old file"
+    );
+    csvContent.push(csvRow);
+  }
+
+  // Iterate over changed rows
+  for (const row of comparisonResult.value.changed_rows) {
+    const fieldsChanged = Object.keys(row.changes)
+      .filter((key) => key !== comparisonHeader)
+      .join(", ");
+
+    // Create two rows: one for the old value and one for the new value
+    const oldCsvRow = createCsvRow(
+      row.id,
+      Object.fromEntries(
+        Object.entries(row.changes).map(([key, value]) => [key, Array.isArray(value) ? value[0] : value])
+      ),
+      oldFileName,
+      `Changed fields: ${fieldsChanged}`
+    );
+
+    const newCsvRow = createCsvRow(
+      row.id,
+      Object.fromEntries(
+        Object.entries(row.changes).map(([key, value]) => [key, Array.isArray(value) ? value[1] : value])
+      ),
+      newFileName,
+      `Changed fields: ${fieldsChanged}`
+    );
+
+    csvContent.push(oldCsvRow);
+    csvContent.push(newCsvRow);
+  }
+
+  // Convert CSV content to a single string
+  csvContent = csvContent.join("\n");
+
+  try {
+    const filePath = await save({
+      filters: [{ name: "", extensions: ["csv"] }],
```

Comment:
left this blank, but it's still defaulting to 'untitled' - wasn't able to make this work without providing this filters object